packages/dds/tree/src/feature-libraries/editable-tree/editableTree.ts

```diff
@@ -314,15 +314,35 @@ export class NodeProxyTarget extends ProxyTarget<Anchor> {
 		eventName: K,
 		listener: EditableTreeEvents[K],
 	): () => void {
-		assert(eventName === "changing", 0x5b3 /* unexpected eventName */);
-		const unsubscribeFromValueChange = this.anchorNode.on("valueChanging", () => listener());
-		const unsubscribeFromChildrenChange = this.anchorNode.on("childrenChanging", () =>
-			listener(),
-		);
-		return () => {
-			unsubscribeFromValueChange();
-			unsubscribeFromChildrenChange();
-		};
+		switch (eventName) {
+			case "changing": {
+				const unsubscribeFromValueChange = this.anchorNode.on(
+					"valueChanging",
+					(anchorNode: AnchorNode, value: Value) => listener(anchorNode, value),
+				);
+				const unsubscribeFromChildrenChange = this.anchorNode.on(
+					"childrenChanging",
+					(anchorNode: AnchorNode) => listener(anchorNode, undefined),
+				);
+				return () => {
+					unsubscribeFromValueChange();
+					unsubscribeFromChildrenChange();
+				};
+			}
+			case "subtreeChanging": {
+				const unsubscribeFromSubtreeChange = this.anchorNode.on(
+					"subtreeChanging",
+					(anchorNode: AnchorNode) => {
+						return (listener as EditableTreeEvents["subtreeChanging"])(anchorNode);
+					},
```

Comment:
I'd prefer to avoid this `as` cast if practical. I think you can avoid it by:
```suggestion
				const typedListener = listener;
				const unsubscribeFromSubtreeChange = this.anchorNode.on(
					"subtreeChanging",
					(anchorNode: AnchorNode) => typedListener(anchorNode)
```

or even:
I'd prefer to avoid this `as` cast if practical. I think you can avoid it by:
```suggestion
				const typedListener = listener;
				const unsubscribeFromSubtreeChange = this.anchorNode.on(
					"subtreeChanging",
					typedListener,
```

Reply:
Wonderful! I confirm, the additional declaration actually tweaks the compiler. Thank you for the insight!

Reply:
Actually appeared only to compile properly in the editor, probably some compilation lag... __Fails on the build.__ I am open to other suggestions as I also don't like the above trick. I know that following code also compiles:
```
const unsubscribeFromSubtreeChange = this.anchorNode.on(
  "subtreeChanging",
  (anchorNode: AnchorNode) => listener(anchorNode, undefined),
);
```
but the vscode hover call hints are off (ie misleading). Maybe Typescript version upgrade ?

Reply:
As unfortunately your suggestion did not work, eliminated the cast as outlined in my prior comment. Is this fine ?

Reply:
As I noted in a top level comment on this PR, this is fine, but the reasons are complicted.